src/Illuminate/Http/Concerns/InteractsWithInput.php

```diff
@@ -307,6 +308,30 @@ public function collect($key = null)
         return collect(is_array($key) ? $this->only($key) : $this->input($key));
     }
 
+    /**
+     * Retrieve input from the request as a Carbon instance.
+     *
+     * @param  string  $key
+     * @param  string|null  $format
+     * @param  string|null  $tz
+     * @param  \Closure|string|null  $default
+     * @return \DateTimeInterface|\Illuminate\Support\Carbon|null
```

Comment:
`DateTimeInterface` is redundant because it's implemented by `Illuminate\Support\Carbon`.

Because `$default` (or its return value if passed a closure) can be mixed, this method should return `mixed` as well. Since this would not be very helpful type-wise, I'd suggest removing the `$default` altogether. The same fallback behaviour can be achieved with a null coalesce operator after `$request->datetime('...')`.

Reply:
Yep, makes sense.